pkg/chunkenc/unordered.go

```diff
@@ -87,21 +89,30 @@ func (hb *unorderedHeadBlock) UncompressedSize() int {
 }
 
 func (hb *unorderedHeadBlock) Reset() {
-	x := newUnorderedHeadBlock()
+	x := newUnorderedHeadBlock(hb.format)
 	*hb = *x
 }
 
+type nsEntry struct {
```

Comment:
That'd conflict with https://github.com/grafana/loki/blob/21df77b51999e36b83503ac7b96c2b26860aa07e/pkg/chunkenc/memchunk.go#L333

I named it after `nsEntry` since it's used inside `nsEntries`